### PR TITLE
Add Windows Release Builds

### PR DESCRIPTION
**Pull request by jkmassel**
## Proposed Changes
Creates signed Windows builds each time a new tag is created.

## Testing Instructions
See https://a8c.slack.com/archives/C06DRMD6VPZ/p1715362399654439, note that each build is downloadable (and signed)

### CODE REVIEW

File: .buildkite/commands/build-for-windows-dev.ps1 [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+& "$PSScriptRoot\prepare-windows-host.ps1"
+
+Write-Host "--- :node: Building App"
+node ./scripts/prepare-dev-build-version.mjs
+If ($LastExitCode -ne 0) { Exit $LastExitCode }
+
+npm run make
+If ($LastExitCode -ne 0) { Exit $LastExitCode }
+
```
Comment (mokagio) [inline .buildkite/commands/build-for-windows-dev.ps1:12]:
```suggestion
npm run make
If ($LastExitCode -ne 0) { Exit $LastExitCode }
```

File: .buildkite/commands/build-for-windows-release.ps1 [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+& "$PSScriptRoot\prepare-windows-host.ps1"
+
+Write-Host "--- :node: Building App"
+node ./scripts/confirm-tag-matches-version.mjs
+If ($LastExitCode -ne 0) { Exit $LastExitCode }
+
+npm run make
+If ($LastExitCode -ne 0) { Exit $LastExitCode }
+
```
Comment (mokagio) [inline .buildkite/commands/build-for-windows-release.ps1:12]:
```suggestion
npm run make
If ($LastExitCode -ne 0) { Exit $LastExitCode }
```

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -64,26 +65,56 @@ steps:
...
+
+      echo "--- :node: Generating Release Manifest"
+      install_npm_packages
+      node ./scripts/generate-releases-manifest.mjs
+
+      echo "--- :fastlane: Distributing Dev Build"
+      install_gems
+      bundle exec fastlane distribute_release_build
+    agents:
+      queue: "mac" #
```
Comment (mokagio) [inline .buildkite/pipeline.yml:135]:
```suggestion
      queue: "mac"
```

File: scripts/generate-releases-manifest.mjs [hunk 4]
```diff
@@ -135,21 +135,34 @@ if ( isDevBuild ) {
 } else {
 	console.log( 'Adding latest release ...' );
 
-	const releaseVersionZipFilename_x64 = `https://cdn.a8c-ci.services/studio/studio-${ process.platform }-x64-v${ version }.app.zip`;
-	const releaseVersionZipFilename_arm64 = `https://cdn.a8c-ci.services/studio/studio-${ process.platform }-arm64-v${ version }.app.zip`;
-	// const releaseVersionZipFilename_win32 = `https://cdn.a8c-ci.services/studio/studio-${ process.platform }-arm64-v${ version }.app.zip`;
+	const releaseVersionZipFilename_mac = `https://cdn.a8c-ci.services/studio/studio-darwin-v${ version }.app.zip`;
+	const releaseVersionZipFilename_x64 = `https://cdn.a8c-ci.services/studio/studio-darwin-x64-v${ version }.app.zip`;
+	const releaseVersionZipFilename_arm64 = `https://cdn.a8c-ci.services/studio/studio-darwin-arm64-v${ version }.app.zip`;
+	const releaseVersionZipFilename_win32 = `https://cdn.a8c-ci.services/studio/studio-win32-v${ version }.exe`;
```
Comment (mokagio) [inline scripts/generate-releases-manifest.mjs]: Interesting mix of camel- and snake- casing.

Did you use it to better visually distinguish the platform?

Did you consider:

```suggestion
	const releaseVersionZipFilenameMac = `https://cdn.a8c-ci.services/studio/studio-darwin-v${ version }.app.zip`;
	const releaseVersionZipFilenameX64 = `https://cdn.a8c-ci.services/studio/studio-darwin-x64-v${ version }.app.zip`;
	const releaseVersionZipFilenameArm64 = `https://cdn.a8c-ci.services/studio/studio-darwin-arm64-v${ version }.app.zip`;
	const releaseVersionZipFilenameWin32 = `https://cdn.a8c-ci.services/studio/studio-win32-v${ version }.exe`;
```

**Review by mokagio — APPROVED** [review]
I haven't manually tested these due to them being build-and-release changes, but all looks good. Only left a few nitpicks.

**Review by wojtekn — APPROVED** [review]

File: scripts/generate-releases-manifest.mjs (hunk 4 above)
Comment (jkmassel) [reply scripts/generate-releases-manifest.mjs]: Addressed in [ed8c467](https://github.com/Automattic/studio/pull/114/commits/ed8c4679f15e554955a7c768421f6597a6da3faf)

**Pull request merged**